### PR TITLE
Uploaders info

### PR DESCRIPTION
**Pull request by prakhar-s**
created an api to fetch uploader details like name,profile pic etc from elasticsearch using authorIds as input

### CODE REVIEW

File: src/main/java/com/strandls/esmodule/models/UploadersInfo.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
+package com.strandls.esmodule.models;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class UploadersInfo {
+
+	String name;
```
Comment (akrudra) [inline src/main/java/com/strandls/esmodule/models/UploadersInfo.java:8]: use access modifier
private

File: src/main/java/com/strandls/esmodule/services/ElasticSearchService.java [hunk 2]
```diff
@@ -28,6 +29,8 @@
  * @author mukund
  */
 public interface ElasticSearchService {
+	
+	List<UploadersInfo> uploaderInfo(String index,String userIds);
```
Comment (akrudra) [inline src/main/java/com/strandls/esmodule/services/ElasticSearchService.java:33]: define access modifier public in the deifination

File: src/main/java/com/strandls/esmodule/services/impl/ElasticSearchServiceImpl.java [hunk 3]
```diff
@@ -562,6 +584,51 @@ public AggregationResponse aggregation(String index, String type, MapSearchQuery
...
+
+					Long authorId = Long.parseLong(String.valueOf(sourceMap.get("author_id")));
+
+					UploadersInfo uploaderInfo = new UploadersInfo(name, pic, authorId);
+
+					result.add(uploaderInfo);
+
+				}
+			} catch (IOException e) { // TODO Auto-generated catch block
+				e.printStackTrace();
```
Comment (akrudra) [inline src/main/java/com/strandls/esmodule/services/impl/ElasticSearchServiceImpl.java:624]: remove printstackstrace with logger, and use exception catch at the last

**Review by akrudra — CHANGES_REQUESTED** [review]
Please do proper formatting and remove unused imports and warning.

File: src/main/java/com/strandls/esmodule/models/UploadersInfo.java (hunk 1 above)
Comment (prakhar-s) [reply src/main/java/com/strandls/esmodule/models/UploadersInfo.java]: added private

File: src/main/java/com/strandls/esmodule/services/ElasticSearchService.java (hunk 2 above)
Comment (prakhar-s) [reply src/main/java/com/strandls/esmodule/services/ElasticSearchService.java]: added public

File: src/main/java/com/strandls/esmodule/services/impl/ElasticSearchServiceImpl.java (hunk 3 above)
Comment (prakhar-s) [reply src/main/java/com/strandls/esmodule/services/impl/ElasticSearchServiceImpl.java]: added logger

**Review by akrudra — APPROVED** [review]
seems good, make sure to check for unused imports and warnings and  code formatting

File: src/main/java/com/strandls/esmodule/ApiConstants.java [hunk 4]
```diff
@@ -47,7 +47,8 @@ private ApiConstants() {}
 	public static final String FORCEUPDATE = "forceupdateindex";
 	public static final String FETCHINDEX="fetchindex";
 	public static final String USERINFO = "/userInfo";
-	
+	public static final String UPLOADERSINFO="/uploadersInfo";
```
Comment (akrudra) [inline src/main/java/com/strandls/esmodule/ApiConstants.java:50]: please put proper spacing, format the code using cntrl+shift+f

File: src/main/java/com/strandls/esmodule/Constants.java [hunk 5]
```diff
@@ -17,5 +17,8 @@ private Constants() {}
 	public static final String AUTHOR_NAME = "author_name";
 	public static final String GROUP_BY_AUTHOR = "group_by_author";
 	public static final String ACTIVITY_SCORE = "activity_score";
+	public static final String MVR_SCIENTIFIC_NAME="max_voted_reco.scientific_name.keyword";
```
Comment (akrudra) [inline src/main/java/com/strandls/esmodule/Constants.java:20]: please apply code formating

File: src/main/java/com/strandls/esmodule/controllers/ESController.java [hunk 6]
```diff
@@ -89,6 +91,48 @@
...
+
+			return Response.status(Status.OK).entity(result).build();
+		} catch (Exception e) {
+			return Response.status(Status.BAD_REQUEST).build();
+		}
+
+	}
+
+	@GET
+	@Path(ApiConstants.UPLOADERSINFO+"/{index}/{userIds}")
```
Comment (akrudra) [inline src/main/java/com/strandls/esmodule/controllers/ESController.java:117]: please apply code formating

**Review by akrudra — CHANGES_REQUESTED** [review]
Please apply proper spacing and code formatting in all the files.

File: src/main/java/com/strandls/esmodule/Constants.java (hunk 5 above)
Comment (prakhar-s) [reply src/main/java/com/strandls/esmodule/Constants.java]: applied the formatting

File: src/main/java/com/strandls/esmodule/controllers/ESController.java (hunk 6 above)
Comment (prakhar-s) [reply src/main/java/com/strandls/esmodule/controllers/ESController.java]: applied

File: src/main/java/com/strandls/esmodule/ApiConstants.java (hunk 4 above)
Comment (prakhar-s) [reply src/main/java/com/strandls/esmodule/ApiConstants.java]: applied proper spacing

**Review by akrudra — APPROVED** [review]

**Pull request merged**